examples/geomstats_in_pymanopt.py

```diff
@@ -0,0 +1,113 @@
+"""This very simple example demonstrates how geometries from geomstats can be
+used in pymanopt to perform optimization on manifolds. It uses the Riemannian
+steepest descent solver.
+
+The example currently requires installing the pymanopt HEAD from git:
+
+    pip install git+ssh://git@github.com/pymanopt/pymanopt.git
+"""
+
+import functools
+import os
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"  # NOQA
+
+import numpy as np
+from numpy import linalg as la, random as rnd
+from pymanopt import Problem
+from pymanopt.manifolds.manifold import Manifold
+from pymanopt.solvers import SteepestDescent
+
+from geomstats.geometry.hypersphere import Hypersphere
+
+
+def squeeze_ndarray(f):
```

Comment:
Very nice, and way shorter than the one that I started to implement in PR #379 . (which implements a decorator's decorator in order to have parameters in the decorator) Could we refactorize this code somehow?

Reply:
I've had a look at doing this via `geomstats.vectorization.decorator`, and was bitten by the fact that the decorator currently doesn't support instance methods (see https://github.com/geomstats/geomstats/pull/379#issuecomment-576250069). Isn't it overkill to use the decorator in this case anway? I feel like we're muddying terminology a bit at this point. The `squeeze_array` decorator here isn't concerned with vectorization, just with the removal of singleton dimensions.